zebra/zebra_rib.c

```diff
@@ -641,15 +641,56 @@ struct route_entry *rib_lookup_ipv4(struct prefix_ipv4 *p, vrf_id_t vrf_id)
 	return NULL;
 }
 
+struct route_entry *rib_lookup(struct prefix *p, vrf_id_t vrf_id)
+{
+	struct route_table *table;
+	struct route_node *rn;
+	struct route_entry *match = NULL;
+	rib_dest_t *dest;
+
+
+	/* Lookup table.  */
+	table = zebra_vrf_table(family2afi(p->family), SAFI_UNICAST, vrf_id);
+	if (!table)
+		return 0;
+
+	rn = route_node_lookup(table, p);
+
+	/* No route for this prefix. */
+	if (!rn)
+		return NULL;
+
+	/* Unlock node. */
+	route_unlock_node(rn);
+	dest = rib_dest_from_rnode(rn);
+
+	if (dest && dest->selected_fib
+	    && !CHECK_FLAG(dest->selected_fib->status, ROUTE_ENTRY_REMOVED))
+		match = dest->selected_fib;
+
+	if (!match)
+		return NULL;
+
+	if (match->type == ZEBRA_ROUTE_CONNECT)
+		return match;
+
+	if (CHECK_FLAG(match->status, ROUTE_ENTRY_INSTALLED))
+		return match;
+
+	return NULL;
+}
+
 /*
  * Is this RIB labeled-unicast? It must be of type BGP and all paths
  * (nexthops) must have a label.
  */
 int zebra_rib_labeled_unicast(struct route_entry *re)
 {
 	struct nexthop *nexthop = NULL;
+	struct zebra_vrf *zvrf = vrf_info_lookup(re->vrf_id);
 
-	if (re->type != ZEBRA_ROUTE_BGP)
+	if ((re->type != ZEBRA_ROUTE_BGP) &&
+	    !zvrf->zebra_mpls_fec_recursive_seamless)
```

Comment:
Again, why this change?

Reply:
Please look at all the places where this function is called. We can't just return true in all of those paths.

Reply:
Hello, I updated the description  with IETF document that highlights the approach proposed by these series of patches (
I hope this might help )